asf_heat_pump_suitability/pipeline/run_scripts/run_make_webmap_geojsons.py

```diff
@@ -0,0 +1,240 @@
+
+
+def check_output_directory(directorystring: str) -> str:
+    """Check if the output directory exists."""
+    uri = urlparse(directorystring)
+    # if s3, test bucket exists
+    if uri.scheme == 's3':
+        s3 = boto3.resource('s3')
+        try:
+            s3.meta.client.head_bucket(Bucket=uri.netloc)
+            return directorystring
+        except:
+            raise OSError(f"Couldn't connect to S3 Bucket: {uri.netloc}, check it exists and is accessible.")
+    elif uri.scheme == "":
+        # assume local file, test if it exists
+        if os.path.isdir(directorystring):
+            return directorystring
+    raise OSError(f"Couldn't connect to {directorystring} check it exists and is accessible.")
+
+
+if __name__ == "__main__":
+    args = parse_arguments()
+
+    # Get Scores file uri logic.
+    if args.scores == "LATEST":
+        # Get latest version of suitability scores.
+        logger.info(f"Getting latest scores file.")
+        file_uri = get_latest_scores_parquet_file_uri()
+    else:
+        logger.info(f"Checking user provided scores file path.")
+        file_uri = get_file_uri(args.scores)
+
+    # check parquet - probably assuming a single-part parquet file (e.g. non-spark-like)
+    filename, file_extension = os.path.splitext(file_uri)
+    if file_extension not in ['.pq', '.parquet', '.pqt']:
+        raise NotImplementedError(f"Scores file must be in parquet format.")
+
+    if args.output_dir:
+        # If an output directory has been specified, check it exists.
+        logger.info(f"Checking user provided output directory.")
+        output_dir = check_output_directory(args.output_dir)
+    else:
+        logger.info(f"Getting output directory from config.")
+        output_dir = config['webmap_data_source']['out_key']
+
+    # Load scores file
+    logger.info(f"Reading suitability scores from {file_uri}")
+    scores = pd.read_parquet(file_uri)
+
+    # Now iterate over the different resolution geojsons to create the joined files.
+    geojson_filestrings = config['webmap_data_source']
+    key_order = (('EW_detailed_lsoa_geojson', 'S_detailed_datazone_geojson'),
+                 ('EW_generalised_lsoa_geojson', 'S_generalised_datazone_geojson'),
+                 ('EW_super_generalised_lsoa_geojson', 'S_super_generalised_datazone_geojson'),
+                 ('EW_ultra_generalised_lsoa_geojson', 'S_ultra_generalised_datazone_geojson'))
+    outfile_name = ['detailed_areas', 'generalised_areas', 'super_generalised_areas', 'ultra_generalised_areas']
+    logger.info(f"Creating output geojsons.")
+    for i, (ew, s) in enumerate(key_order):
+        # Load geometries
+        ew_geojson = gpd.read_file(geojson_filestrings[ew]).loc[:, ['LSOA21CD', 'geometry']].rename(columns={'LSOA21CD': 'area_code'})
+        s_geojson = gpd.read_file(geojson_filestrings[s]).loc[:, ['DataZone', 'geometry']].rename(columns={'DataZone': 'area_code'})
```

Comment:
```suggestion
        ew_gdf = gpd.read_file(geojson_filestrings[ew]).loc[:, ['LSOA21CD', 'geometry']].rename(columns={'LSOA21CD': 'area_code'})
        s_gdf = gpd.read_file(geojson_filestrings[s]).loc[:, ['DataZone', 'geometry']].rename(columns={'DataZone': 'area_code'})
```
here and below these should be `gdf` i think - for readability

Reply:
Have changed these to _gdf